src/Inventory.zig

```diff
@@ -507,69 +510,167 @@ pub const Command = struct { // MARK: Command
-				.durability = std.mem.readInt(i32, data[12..16], .big),
-				.item = null,
-			};
-			if(data.len > 16) {
-				const zon = ZonElement.parseFromString(main.stackAllocator, data[16..]);
-				defer zon.deinit(main.stackAllocator);
-				self.item = try Item.init(zon);
-			}
-			if(self.amount > 0) { // Create
-				if(self.item) |item| {
-					self.inv.ref().item = item;
-				} else if(self.inv.ref().item == null) {
-					return error.Invalid;
-				}
-				if(self.inv.ref().amount +| self.amount > self.inv.ref().item.?.stackSize()) {
-					return error.Invalid;
-				}
-				self.inv.ref().amount += @intCast(self.amount);
-			} else if(self.amount < 0) { // Delete
-				if(self.inv.ref().amount < -self.amount) {
-					return error.Invalid;
-				}
-				self.inv.ref().amount -= @intCast(-self.amount);
-				if(self.inv.ref().amount == 0) {
-					self.inv.ref().item = null;
+			std.debug.assert(data.len >= 1);
+
+			switch (try deserialize(data)) {
+				.create => |create| {
+					if (create.item) |item| {
+						create.inv.ref().item = item;
+					} else if (create.inv.ref().item == null) {
+						return error.Invalid;
+					}
+
+					if (create.inv.ref().amount +| create.amount > create.inv.ref().item.?.stackSize()) {
+						return error.Invalid;
+					}
+					create.inv.ref().amount += create.amount;
+					
+					create.inv.inv.update();
+				},
+				.delete => |delete| {
+					if (delete.inv.ref().amount < delete.amount) {
+						return error.Invalid;
+					}
+					delete.inv.ref().amount -= delete.amount;
+					if (delete.inv.ref().amount == 0) {
+						delete.inv.ref().item = null;
+					}
+					
+					delete.inv.inv.update();
+				},
+				.useDurability => |durability| {
+					durability.inv.ref().item.?.tool.durability = @intCast(durability.durability);
+					if (durability.inv.ref().item.?.tool.durability == 0) {
+						durability.inv.ref().item = null;
+						durability.inv.ref().amount = 0;
+					}
+					
+					durability.inv.inv.update();
 				}
 			}
+		}
 
-			if(self.durability < 0) { // useDurability
-				self.inv.ref().item.?.tool.durability -|= @intCast(-self.durability);
-				if(self.inv.ref().item.?.tool.durability == 0) {
-					self.inv.ref().item = null;
-					self.inv.ref().amount = 0;
+		pub fn getUsers(self: SyncOperation, allocator: NeverFailingAllocator) []*main.server.User {
+			switch (self) {
+				.create => |create| {
+					return allocator.dupe(*main.server.User, Sync.ServerSide.inventories.items[create.inv.inv.id].users.items);
+				},
+				.delete => |delete| {
+					return allocator.dupe(*main.server.User, Sync.ServerSide.inventories.items[delete.inv.inv.id].users.items);
+				},
+				.useDurability => |durability| {
+					return allocator.dupe(*main.server.User, Sync.ServerSide.inventories.items[durability.inv.inv.id].users.items);
 				}
 			}
+		}
+
+
+		pub fn ignoreSource(self: SyncOperation) bool {
+			return switch (self) {
+				.create, .delete, .useDurability => true,
+			};
+		}
```

Comment:
I would like you to move this one function into the health PR. I'm not yet convinced if this is the right solution.